testsuite/mixed-domain/src/test/java/org/jboss/as/test/integration/domain/mixed/MixedDomainTestSupport.java

```diff
@@ -0,0 +1,138 @@
+                "slave-config/host-slave.xml", dir.getAbsolutePath());
+    }
+
+    public void start() {
+        try {
+            //Start the master in admin only  and reconfigure the domain with what
+            //we want to test in the mixed domain and have the DomainAdjuster
+            //strip down the domain model to something more workable. The domain
+            //adjusters will also make adjustments for the legacy version being
+            //tested.
+            DomainLifecycleUtil masterUtil = getDomainMasterLifecycleUtil();
+            masterUtil.getConfiguration().setAdminOnly(true);
+            masterUtil.start();
+            DomainAdjuster.adjustForVersion(masterUtil.getDomainClient(), version);
+
+            //Now reload the master in normal mode
+            masterUtil.executeAwaitConnectionClosed(Util.createEmptyOperation("reload", PathAddress.pathAddress(HOST, "master")));
+            masterUtil.connect();
+            masterUtil.awaitHostController(System.currentTimeMillis());
+
+            //Start the slaves
+            DomainLifecycleUtil slaveUtil = getDomainSlaveLifecycleUtil();
+            if (slaveUtil != null) {
+                //slaveUtil.getConfiguration().addHostCommandLineProperty("-agentlib:jdwp=transport=dt_socket,address=8787,server=y,suspend=y");
+                slaveUtil.start();
+            }
+
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private static String copyDomainFile() throws Exception {
+
+        final File originalDomainXml = loadFile("target", "jbossas", "domain", "configuration", "domain.xml");
+        final File targetDirectory = createDirectory("target", "test-classes", "copied-master-config");
+        final File copiedDomainXml = new File(targetDirectory, "domain.xml");
+        if (copiedDomainXml.exists()) {
+            Assert.assertTrue(copiedDomainXml.delete());
+        }
+        final InputStream in = new BufferedInputStream(new FileInputStream(originalDomainXml));
+        try {
+            final OutputStream out = new BufferedOutputStream(new FileOutputStream(copiedDomainXml));
+            try {
+                byte[] bytes = new byte[1024];
+                int len = in.read(bytes);
+                while (len != -1) {
+                    out.write(bytes, 0, len);
+                    len = in.read(bytes);
+                }
+            } finally {
+                safeClose(out);
+            }
+        } finally {
+            safeClose(in);
+        }
+        return "copied-master-config/domain.xml";
+    }
+
+    private static File loadFile(String... parts) {
+        File file = new File(".");
+        for (String part : parts) {
+            file = new File(file, part);
+            Assert.assertTrue(file.getAbsolutePath() + " does not exist", file.exists());
+        }
+        return file;
+    }
+
+    private static File createDirectory(String...parts) {
+        File file = new File(".");
```

Comment:
pretty much whole method can be replaced by 
Files.createDirectories(Paths.get(parts))